### PR TITLE
Added code for RemoveStep command and parser, Add/RemoveTag command and parser

### PR DESCRIPTION
**Pull request by hans555**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.recipe.Recipe;
+import seedu.address.model.step.Step;
+import seedu.address.model.step.UniqueStepList;
 
 /**
- * todo
+ * Removes a step to a recipe.
```
Comment (gohkh) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java]: should be "remove step from"

**Review by gohkh — APPROVED** [review]
Just need to fix build errors.

**Review by teo-jun-xiong — APPROVED** [review]
LGTM, need to fix the check style errors

**Review by tzihiang — APPROVED** [review]
LGTM, just need to adjust AppVeyor build issues.

**Review by teo-jun-xiong — APPROVED** [review]
LGTM

**Review by shanecsj — APPROVED** [review]

**Review by tzihiang — APPROVED** [review]

File: src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        requireNonNull(recipeIndex);
+        requireNonNull(toAdd);
+        this.recipeIndex = recipeIndex;
+        this.toAdd = toAdd;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Recipe> lastShownList = model.getFilteredRecipeList();
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java]: getFilteredCookbookRecipeList

File: src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        Set<Tag> targetTagSet = recipeToEdit.getTags();
+
+        if (!targetTagSet.add(toAdd)) {
+            throw new CommandException(Messages.MESSAGE_DUPLICATE_TAG);
+        }
+
+        EditRecipeDescriptor editRecipeDescriptor = new EditRecipeDescriptor();
+        editRecipeDescriptor.setTags(targetTagSet);
+        Recipe editedRecipe = EditRecipeDescriptor.createEditedRecipe(recipeToEdit, editRecipeDescriptor);
+        model.setRecipe(recipeToEdit, editedRecipe);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java]: setCookbookRecipe

File: src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        model.updateFilteredRecipeList(PREDICATE_SHOW_ALL_RECIPES);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeAddTagCommand.java]: updateFilteredCookbookRecipeList

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java [hunk 5]
```diff
@@ -26,6 +33,36 @@ public RecipeRemoveStepCommand(Index recipeIndex, Index stepIndex) {
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return null;
+        requireNonNull(model);
+        List<Recipe> lastShownList = model.getFilteredRecipeList();
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java]: getFilteredCookbookRecipeList

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java [hunk 6]
```diff
@@ -26,6 +33,36 @@ public RecipeRemoveStepCommand(Index recipeIndex, Index stepIndex) {
...
+            // ensure the step index is valid
+            throw new CommandException((Messages.MESSAGE_INVALID_STEP_DISPLAYED_INDEX));
+        }
+
+        Step toRemove = targetStepList.remove(stepIndex);
+
+        EditRecipeDescriptor editRecipeDescriptor = new EditRecipeDescriptor();
+        editRecipeDescriptor.setSteps(targetStepList);
+        Recipe editedRecipe = EditRecipeDescriptor.createEditedRecipe(recipeToEdit, editRecipeDescriptor);
+        model.setRecipe(recipeToEdit, editedRecipe);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java]: setCookbookRecipe

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java [hunk 7]
```diff
@@ -26,6 +33,36 @@ public RecipeRemoveStepCommand(Index recipeIndex, Index stepIndex) {
...
+        model.updateFilteredRecipeList(PREDICATE_SHOW_ALL_RECIPES);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveStepCommand.java]: updateFilteredCookbookRecipeList

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        requireNonNull(recipeIndex);
+        requireNonNull(toRemove);
+        this.recipeIndex = recipeIndex;
+        this.toRemove = toRemove;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Recipe> lastShownList = model.getFilteredRecipeList();
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java]: getFilteredCookbookRecipeList

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        Set<Tag> targetTagSet = recipeToEdit.getTags();
+
+        if (!targetTagSet.remove(toRemove)) {
+            throw new CommandException((Messages.MESSAGE_INVALID_TAG));
+        }
+
+        EditRecipeDescriptor editRecipeDescriptor = new EditRecipeDescriptor();
+        editRecipeDescriptor.setTags(targetTagSet);
+        Recipe editedRecipe = EditRecipeDescriptor.createEditedRecipe(recipeToEdit, editRecipeDescriptor);
+        model.setRecipe(recipeToEdit, editedRecipe);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java]: setCookbookRecipe

File: src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        model.updateFilteredRecipeList(PREDICATE_SHOW_ALL_RECIPES);
```
Comment (teo-jun-xiong) [inline src/main/java/seedu/address/logic/commands/recipe/RecipeRemoveTagCommand.java]: updateFilteredCookbookRecipeList

**Pull request merged**